storage/block_devices_ops.go

```diff
@@ -2076,6 +2063,116 @@ func GetAdvancedPartitions(medias []*BlockDevice) []string {
 	return results
 }
 
+// setBootPartition is a helper function to PrepareInstallationMedia
+// Looks through all of the installation media to determine which
+// partition will be the one from which the install boots
+// either an explicit /boot or / (root) in legacy mode
+func setBootPartition(medias []*BlockDevice, mediaOpts MediaOpts, dryRun *[]string) error {
+	const (
+		bootStyleDefault = "boot"
+		bootStyleLegacy  = "legacy_boot"
+	)
+
+	logFormatError := func(format string, vargs ...interface{}) string {
+		mesg := utils.Locale.Get(format, vargs...)
+		log.Error(mesg)
+
+		if dryRun != nil {
+			*dryRun = append(*dryRun, mesg)
+		}
+
+		return mesg
+	}
+
+	style := bootStyleDefault
+	var bootParent, bootBlockDevice, rootParent, rootBlockDevice *BlockDevice
+
+	// Check if there is a bootable partition
+	// Clear Linux OS only supports booting from a top level
+	// block device; RAID, LVM, encryption, etc are not supported
+	for _, bd := range medias {
+		for _, curr := range bd.Children {
+			// We have the standard /boot partition
+			if curr.MountPoint == "/boot" {
+				if bootBlockDevice != nil {
+					return errors.Errorf(logFormatError("Found multiple %s partition names", curr.MountPoint))
+				}
+				bootParent = bd
```

Comment:
I would suggest that we use something like the https://github.com/clearlinux/clr-installer/blob/master/storage/block_devices_filters.go#L20 function to find blockdevices that meet a certain condition....I am bringing in a few more that help in my make destructive patch.

I think your condition wont be satisfied with it alone. I have seen at many places where we want parent and a child returned if certain condition is met. It Simplifies the code and prevents repetition

Reply:
Currently this is the only use case for needing both the child and parent. Find the boot partition is a special case.